draft-aegis-mimi-arch-latest.md

```diff
@@ -209,39 +215,35 @@ according to the rules in {{!RFC5280}} using the trust roots agreed upon by the
 (TODO: Some sort of extension dealing with trust root negotiation).
 
 ~~~
-struct {
-    uint64 msg_timestamp;
-} X509ApplicationContext;
+X509ApplicationContext
+{
+    "msg_timestamp": uint64
+}
 
-struct {
-    uint8 start;
-    uint8 end<0...255>;
-} X509DeviceHandleRange
+X509DeviceHandleRange
+{
+    "start": uint8,
+    "end": uint8
+}
 
-struct {
-    X509DeviceHandleRange device_handle_range;
-    select (IdentityConfiguration.supports_accounts) {
-        case true:
-            uint8 account_handle_offset;
-        case false:
-            struct {}
-    }
-    select (IdentityConfiguration.supports_domains) {
-        case true:
-            uint8 domain_name_offset;
-        case false:
-            struct {}
-    }
-} X509Parameters
+X509Parameters
+{      // required
+    "device_handle_range": X509DeviceHandleRange,
+       // required when IdentityConfiguration.support_accounts == true
+    "account_handle_offset": uint8,
+       // required when IdentityConfiguration.support_domains == true    
+    "domain_name_offset": uint8
+}
 ~~~
 
-A given parameter set `IdentityConfiguration conf` with custom X.509 Identity
-Provider parameters `X509Parameters params` is valid if the following
-conditions are all met. The device handle range MUST not end before it
+A given `IdentityConfiguration` parameter set with custom X.509 Identity
+Provider settings `"parameters": object<X509Parameters>` are valid if the
```

Comment:
What does this notation `"parameters": object<X509Parameters>` mean?

Reply:
parameters is declared originally as being an unknown JSON object. Now we're making it concrete we know the type of object it is: a X509Parameters object.

I dont really know how to do this more cleanly though... @tomleavy or @mulmarta, feel free to change if you know clearer way.

Reply:
Yes that is correct @mulmarta it's generically injected so we can use the type directly. JavaScript has no concepts of types at all so either we place whatever object where we need it, or we specify a duck type interface you must minimally conform to

Reply:
OK. done.